SA/design_concept/content/tool_research/agda.typ

```diff
@@ -0,0 +1,33 @@
+= Agda
+
+Even though Agda
+#footnote("https://wiki.portal.chalmers.se/agda/pmwiki.php") isn't a visual
+programming environment, the tools for it have been designed in an
+interactive way and are of interest for this project.
+
+The basis for these tools is a language-server, which in combination with
+the powerful checker and editor extensions, allows to not only verify the
+correctness of the code, but also to check incomplete programs. Additional
+context, such as checker errors or incomplete expressions, is then provided
+in a separate window inside the editor. 
+
+Using this tooling it is, for example, possible to insert a hole in the
+code, called a _metavariable_, which the checker will detect and display
+alongside the expected type in the context-window, as shown below.
+
+#figure(
+  image("../../static/agda-context-window.png", width: 70%),
```

Comment:
I think the actual image is currently missing.

Reply:
added